### PR TITLE
Simple Structures

### PR DESCRIPTION
**Pull request by APickledWalrus**
### Description
This PR adds support for registering Structure that do not have sections. That is, they are based on SimpleNodes rather than SectionNodes. They simply look like:
```
# no end-of-line colon below means it is simple!
do something cool

# this is a regular structure
command /foo:
  trigger:
    ...
```

A Structure may be registered this way using `Skript#registerSimpleStructure`.

An API change was required that makes `EntryContainer` nullable in Structure's init. This will not break any existing code as for non-simple Structures, it is guaranteed that EntryContainer is NotNull.

`Structure#getEntryContainer` has also been deprecated. This method's existence was previously justified for internal reasons, but those reasons no longer exist (design has improved), so neither will this method in the future. Developers are encouraged to save a reference to the entry container during initialization if they need it during the loading period.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:**
- https://github.com/SkriptLang/Skript/issues/5318 (non-closing)

### CODE REVIEW

**Review by AyhamAl-Ali — APPROVED** [review]
Good work Pickle! LGTM :rocket:

File: src/main/java/org/skriptlang/skript/lang/structure/Structure.java [hunk 1]
```diff
@@ -105,10 +114,15 @@ public final boolean init(Expression<?>[] expressions, int matchedPattern, Kleen
 		assert structureInfo != null;
 		EntryValidator entryValidator = structureInfo.entryValidator;
 
-		if (entryValidator == null) { // No validation necessary, the structure itself will handle it
-			entryContainer = EntryContainer.withoutValidator(structureData.sectionNode);
-		} else { // Okay, now it's time for validation
-			EntryContainer entryContainer = entryValidator.validate(structureData.sectionNode);
+		if (structureData.node instanceof SimpleNode) {
```
[the remark this answers is not in the document]
Comment (APickledWalrus) [reply src/main/java/org/skriptlang/skript/lang/structure/Structure.java:117]: sure

File: src/main/java/org/skriptlang/skript/lang/structure/Structure.java [hunk 2]
```diff
@@ -173,14 +196,19 @@ public String toString() {
...
 	@Nullable
-	public static Structure parse(String expr, SectionNode sectionNode, @Nullable String defaultError, Iterator<? extends StructureInfo<? extends Structure>> iterator) {
-		ParserInstance.get().getData(StructureData.class).sectionNode = sectionNode;
+	public static Structure parse(String expr, Node node, @Nullable String defaultError, Iterator<? extends StructureInfo<? extends Structure>> iterator) {
+		if (!(node instanceof SimpleNode) && !(node instanceof SectionNode)) {
+			throw new IllegalArgumentException("only simple or section nodes may be parsed as a structure");
+		}
+		ParserInstance.get().getData(StructureData.class).node = node;
 
+		if (node instanceof SimpleNode) // only allow simple structures for simple nodes
```
Comment (APickledWalrus) [inline src/main/java/org/skriptlang/skript/lang/structure/Structure.java:210]: simple structures need to be blocked for SectionNodes... oops!

File: src/main/java/ch/njol/skript/ScriptLoader.java [hunk 3]
```diff
@@ -530,19 +532,19 @@ private static CompletableFuture<ScriptInfo> loadScripts(List<Config> configs, O
 					pairs.removeIf(pair -> {
 						Structure structure = pair.getSecond();
 
-						parser.setActive(pair.getFirst().getFirst());
+						parser.setActive(pair.getFirst().script);
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/ScriptLoader.java:535]: why are we constantly getFirst-ing

File: src/main/java/ch/njol/skript/structures/StructEvent.java [hunk 4]
```diff
@@ -45,7 +45,7 @@ public class StructEvent extends Structure {
 
 	@Override
 	@SuppressWarnings("ConstantConditions")
-	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult, EntryContainer entryContainer) {
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult, @Nullable EntryContainer entryContainer) {
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/structures/StructEvent.java:48]: Line 68 can now throw an NPE.

File: src/main/java/org/skriptlang/skript/lang/structure/Structure.java [hunk 5]
```diff
@@ -173,14 +196,19 @@ public String toString() {
...
+		return parse(expr, node, defaultError, Skript.getStructures().iterator());
 	}
 
 	@Nullable
-	public static Structure parse(String expr, SectionNode sectionNode, @Nullable String defaultError, Iterator<? extends StructureInfo<? extends Structure>> iterator) {
-		ParserInstance.get().getData(StructureData.class).sectionNode = sectionNode;
+	public static Structure parse(String expr, Node node, @Nullable String defaultError, Iterator<? extends StructureInfo<? extends Structure>> iterator) {
+		if (!(node instanceof SimpleNode) && !(node instanceof SectionNode)) {
+			throw new IllegalArgumentException("only simple or section nodes may be parsed as a structure");
+		}
```
Comment (sovdeeth) [inline src/main/java/org/skriptlang/skript/lang/structure/Structure.java:207]:
```suggestion
		if (!(node instanceof SimpleNode) && !(node instanceof SectionNode))
			throw new IllegalArgumentException("only simple or section nodes may be parsed as a structure");
```

**Review by sovdeeth — CHANGES_REQUESTED** [review]
looking good!

File: src/main/java/ch/njol/skript/structures/StructEvent.java (hunk 4 above)
Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/structures/StructEvent.java:48]: It's not a simple structure, so it can't be null, but i will add an assertion

**Review by sovdeeth — APPROVED** [review]

**Review by Moderocky — APPROVED** [review]

**Pull request merged**